lighthouse-cli/test/cli/run-test.js

```diff
@@ -19,42 +19,73 @@ const fastConfig = {
+  describe('LH round trip', () => {
+    /** @type {LH.RunnerResult} */
+    let passedResults;
     const filename = path.join(process.cwd(), 'run.ts.results.json');
-    const timeoutFlag = `--max-wait-for-load=${9000}`;
-    const flags = getFlags(`--output=json --output-path=${filename} ${timeoutFlag} ${url}`);
-    return run.runLighthouse(url, flags, fastConfig).then(passedResults => {
-      if (!passedResults) {
-        assert.fail('no results');
-        return;
+    /** @type {LH.Result} */
+    let fileResults;
+
+    beforeAll(async () => {
+      const url = 'chrome://version';
+      const timeoutFlag = `--max-wait-for-load=${9000}`;
+      const pluginsFlag = '--plugins=lighthouse-plugin-simple';
+
+      // eslint-disable-next-line max-len
+      const flags = getFlags(`--output=json --output-path=${filename} ${pluginsFlag} ${timeoutFlag} ${url}`);
+
+      const rawResult = await run.runLighthouse(url, flags, fastConfig);
+
+      if (!rawResult) {
+        return assert.fail('no results');
       }
+      passedResults = rawResult;
 
-      const {lhr} = passedResults;
       assert.ok(fs.existsSync(filename));
-      /** @type {LH.Result} */
-      const results = JSON.parse(fs.readFileSync(filename, 'utf-8'));
-      assert.equal(results.audits.viewport.rawValue, false);
+      fileResults = JSON.parse(fs.readFileSync(filename, 'utf-8'));
+    }, 20 * 1000);
+
+    afterAll(() => {
+      fs.unlinkSync(filename);
+    });
+
+    it('returns results that match the saved results', () => {
+      const {lhr} = passedResults;
+      assert.equal(fileResults.audits.viewport.rawValue, false);
 
       // passed results match saved results
-      assert.strictEqual(results.fetchTime, lhr.fetchTime);
-      assert.strictEqual(results.requestedUrl, lhr.requestedUrl);
-      assert.strictEqual(results.finalUrl, lhr.finalUrl);
-      assert.strictEqual(results.audits.viewport.rawValue, lhr.audits.viewport.rawValue);
+      assert.strictEqual(fileResults.fetchTime, lhr.fetchTime);
+      assert.strictEqual(fileResults.requestedUrl, lhr.requestedUrl);
+      assert.strictEqual(fileResults.finalUrl, lhr.finalUrl);
+      assert.strictEqual(fileResults.audits.viewport.rawValue, lhr.audits.viewport.rawValue);
       assert.strictEqual(
-          Object.keys(results.audits).length,
+          Object.keys(fileResults.audits).length,
           Object.keys(lhr.audits).length);
-      assert.deepStrictEqual(results.timing, lhr.timing);
-      assert.ok(results.timing.total !== 0);
+      assert.deepStrictEqual(fileResults.timing, lhr.timing);
+    });
+
+    it('includes timing information', () => {
+      assert.ok(passedResults.lhr.timing.total !== 0);
+    });
 
-      assert.equal(results.configSettings.channel, 'cli');
+    it('correctly sets the channel', () => {
+      assert.equal(passedResults.lhr.configSettings.channel, 'cli');
+    });
 
-      fs.unlinkSync(filename);
+    it('merged the plugin into the config', () => {
```

Comment:
this is well beyond just stretching the "unit" part of "unit tests", but this check has trivial overhead and until we have a smoketest for plugins I'd like to leave this in